### ISSUE TITLE
System.ArgumentOutOfRangeException when pasting in VS2019

### ISSUE THREAD

**Issue by amoldeshpande** [VsVim/VsVim#2591]
**Describe the bug**
Pasting from clipboard causes messagebox popups and exception message
**To Reproduce**
Copy a file's text with ^A ^C, paste into another window with ^V (edited to add: hitting escape after the paste is what actually produces the popup MB)

**Expected behavior**
Peaceful, easy feeling

**Environment (please complete the following information):**

- Visual Studio version: 2019
- VsVim version: 2.6.0.0
- Programming Language: C++
- Check(Type 'x') any that are installed:
    - [ ] ReSharper
    - [ ] Visual Assist

Activitylog:
```
  <entry>
    <record>1806</record>
    <time>2019/06/02 16:42:10.838</time>
    <type>Error</type>
    <source>Editor or Editor Extension</source>
    <description>
System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values.&#x000D;&#x000A;
Parameter name: span&#x000D;&#x000A; 
  at Microsoft.VisualStudio.Text.SnapshotSpan..ctor(ITextSnapshot snapshot, Span span)&#x000D;&#x000A;
   at Vim.Modes.Insert.TextChangeTracker.get_EffectiveChange()&#x000D;&#x000A;
   at Vim.Modes.Insert.TextChangeTracker.Vim-Modes-Insert-ITextChangeTracker-get_EffectiveChange()&#x000D;&#x000A;
   at Vim.Modes.Insert.InsertMode.OnCaretPositionChanged(CaretPositionChangedEventArgs args)&#x000D;&#x000A;
   at &lt;StartupCode$Vim-Core&gt;.$Modes_Insert_InsertMode.-ctor@295-31.Invoke(CaretPositionChangedEventArgs args)&#x000D;&#x000A;
   at Microsoft.FSharp.Control.CommonExtensions.SubscribeToObservable@2172.System-IObserver`1-OnNext(T value)&#x000D;&#x000A;
   at Microsoft.FSharp.Control.ObservableModule.succeed@2704[TResult](IObserver`1 observer, FSharpOption`1 _arg1)&#x000D;&#x000A;
   at Microsoft.FSharp.Core.CompilerServices.RuntimeHelpers.h@349.Invoke(Object _arg1, TArgs args)&#x000D;&#x000A;
   at &lt;StartupCode$Vim-Core&gt;.$Modes_Insert_InsertMode.-ctor@293-35.Invoke(Object delegateArg0, CaretPositionChangedEventArgs delegateArg1)&#x000D;&#x000A;
   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.RaiseEvent[TArgs](Object sender, EventHandler`1 eventHandlers, TArgs args)&#x000D;&#x000A;
--- End of stack trace from previous location where exception was thrown ---&#x000D;&#x000A;
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
    </description>
  </entry>
```

This may be another manifestation of #2067 but I'm filing just in case it's not.

**Comment by ricksladkey** [comment]
Thanks for filing this issue because it is not another manifestation of the "normal" clipboard problem. I can reproduce this error and I believe that it requires a buffer that does not end in a final newline.

**Comment by ricksladkey** [comment]
Well, I take that back. I cannot reproduce this specific problem.

@amoldeshpande, if you can reproduce this at will, can you collect some data for me?

- Turn on the output window: Tools -> Options -> VsVim -> Defaults -> Output Window -> True
- Open the output window: View -> Output
- Select VsVim output: Show output from -> VsVim
- Perform the steps to reproduce the exception
- Copy all the output in the VsVim output window
- Paste the output into a comment on this issue

You should get a lot of output, specifically some lines like this:

```
VsVim OnTextChange: 1
VsVim OnTextChange: change 0
VsVim OnTextChange: old = '', new = 'abc^M^Jdef^M^J'
VsVim OnTextChange: old = '[0..0)', new = '[0..10)'
VsVim OnTextChange: caret position = 0
```

**Comment by amoldeshpande** [comment]
The exception is not something I've seen more than once. the messagebox is consistent. Note that after you paste, you need to scroll to the bottom of the text and hit escape. If you scroll back up, the 
messagebox goes away.


`VsVim AsyncTagger::RaiseTagsChanged 0 - 0
VsVim MarkGlyphTagger: Glyph Pairs
VsVim MarkGlyphTagger: 20 -> ']
VsVim MarkGlyphTagger: 5 -> [a
VsVim AsyncTagger::RaiseTagsChanged 0 - 21
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: tag 5 [a
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: tag 20 ']
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim ImeCoordinator: in mode = None turning IME Off
VsVim AsyncTagger::GetTags 0 - 21
VsVim AsyncTagger Background New 0 - 21
VsVim AsyncTagger::GetTags 0 - 0
VsVim AsyncTagger::GetTags 0 - 21
VsVim AsyncTagger Background Existing 0 - 21
VsVim AsyncTagger::RaiseTagsChanged 0 - 21
VsVim AsyncTagger::GetTags 0 - 21
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VimKeyProcessor::KeyDown LeftCtrl None Control
VsVim VimKeyProcessor::KeyDown Handled = False
VsVim OnCaretPositionChanged: old = |?_v0_658_' '+0, new = |?_v0_667_'<end>'+0
VsVim MarkGlyphTagger: Glyph Pairs
VsVim MarkGlyphTagger: 20 -> ']
VsVim MarkGlyphTagger: 5 -> [a
VsVim MarkGlyphTagger: 0 -> <
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: tag 0 <
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: tag 5 [a
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: tag 20 ']
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim AsyncTagger::GetTags 0 - 21
VsVim VimBufferCoordinator::PreviewKeyUp A
VsVim VimKeyProcessor::KeyUp A Control
VsVim VimKeyProcessor::KeyUp Handled = False
VsVim VimBufferCoordinator::KeyUp A
VsVim VimBufferCoordinator::PreviewKeyUp C
VsVim VimKeyProcessor::KeyUp C Control
VsVim VimKeyProcessor::KeyUp Handled = False
VsVim VimBufferCoordinator::KeyUp C
VsVim VimBufferCoordinator::PreviewKeyUp LeftCtrl
VsVim VimKeyProcessor::KeyUp LeftCtrl None
VsVim VimKeyProcessor::KeyUp Handled = False
VsVim VimBufferCoordinator::KeyUp LeftCtrl
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim AsyncTagger::RaiseTagsChanged 0 - 21
VsVim MarkGlyphTagger: Glyph Pairs
VsVim MarkGlyphTagger: 0 -> '^
VsVim AsyncTagger::RaiseTagsChanged 0 - 0
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: tag 0 '^
VsVim MarkGlyphTagger::GetTags: starting...
VsVim ImeCoordinator: in mode = Insert turning IME Off
VsVim AsyncTagger::GetTags 0 - 21
VsVim AsyncTagger::GetTags 0 - 0
VsVim AsyncTagger Background New 0 - 0
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VimKeyProcessor::KeyDown LeftCtrl None Control
VsVim VimKeyProcessor::KeyDown Handled = False
VsVim VsCommandTarget::QueryStatus Paste Key.None: 5efc7975-14bc-11cf-9b2b-00aa00573819 26
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::Exec Paste Key.None: 5efc7975-14bc-11cf-9b2b-00aa00573819 26
VsVim OnTextChange: 1
VsVim OnTextChange: change 0
VsVim OnTextChange: old = '', new = 'using System;
using System.Drawing;

namespace libVT100
{
   public enum DeviceStatus
   {
      Unknown = -1,
      Ok = 0,
      Failure = 3,
   }
   
   public interface IVT100DecoderClient : IAnsiDecoderClient
   {
      string GetDeviceCode ( IVT100Decoder _decoder );
      DeviceStatus GetDeviceStatus ( IVT100Decoder _decoder );
      /// <summary>Resize the terminal window to _size (given in characters).</summary>
      void ResizeWindow ( IVT100Decoder _decoder, Size _size );
      /// <summary>Move the terminal window to _size (given in pixels).</summary>
      void MoveWindow ( IVT100Decoder _decoder, Point _position );
   }
}
'
VsVim OnTextChange: old = '[14..14)', new = '[14..681)'
VsVim OnTextChange: caret position = ?_v19_14_'<end>'
VsVim MarkGlyphTagger: Glyph Pairs
VsVim MarkGlyphTagger: 0 -> '^
VsVim MarkGlyphTagger: 22 -> .
VsVim MarkGlyphTagger::TextBufferChanged V20 (r20)
VsVim MarkGlyphTagger: Glyph Pairs
VsVim MarkGlyphTagger: 0 -> '^
VsVim MarkGlyphTagger: 22 -> .
VsVim AsyncTagger::GetTags 1 - 22
VsVim AsyncTagger Background New 1 - 22
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: tag 22 .
VsVim MarkGlyphTagger::GetTags: starting...
VsVim OnTextChange: 28
VsVim OnTextChange: change 0
VsVim OnTextChange: old = '   ', new = '	'
VsVim OnTextChange: old = '[77..80)', new = '[77..78)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 1
VsVim OnTextChange: old = '   ', new = '	'
VsVim OnTextChange: old = '[106..109)', new = '[104..105)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 2
VsVim OnTextChange: old = '      ', new = '		'
VsVim OnTextChange: old = '[112..118)', new = '[108..110)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 3
VsVim OnTextChange: old = '      ', new = '		'
VsVim OnTextChange: old = '[133..139)', new = '[125..127)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 4
VsVim OnTextChange: old = '      ', new = '		'
VsVim OnTextChange: old = '[148..154)', new = '[136..138)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 5
VsVim OnTextChange: old = '   ', new = '	'
VsVim OnTextChange: old = '[168..171)', new = '[152..153)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 6
VsVim OnTextChange: old = '   ', new = ''
VsVim OnTextChange: old = '[174..177)', new = '[156..156)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 7
VsVim OnTextChange: old = '   ', new = '	'
VsVim OnTextChange: old = '[179..182)', new = '[158..159)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 8
VsVim OnTextChange: old = '   ', new = '	'
VsVim OnTextChange: old = '[241..244)', new = '[218..219)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 9
VsVim OnTextChange: old = '      ', new = '		'
VsVim OnTextChange: old = '[247..253)', new = '[222..224)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 10
VsVim OnTextChange: old = ' ', new = ''
VsVim OnTextChange: old = '[273..274)', new = '[244..244)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 11
VsVim OnTextChange: old = ' ', new = ''
VsVim OnTextChange: old = '[275..276)', new = '[245..245)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 12
VsVim OnTextChange: old = ' ', new = ''
VsVim OnTextChange: old = '[298..299)', new = '[267..267)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 13
VsVim OnTextChange: old = '      ', new = '		'
VsVim OnTextChange: old = '[303..309)', new = '[271..273)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 14
VsVim OnTextChange: old = ' ', new = ''
VsVim OnTextChange: old = '[337..338)', new = '[301..301)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 15
VsVim OnTextChange: old = ' ', new = ''
VsVim OnTextChange: old = '[339..340)', new = '[302..302)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 16
VsVim OnTextChange: old = ' ', new = ''
VsVim OnTextChange: old = '[362..363)', new = '[324..324)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 17
VsVim OnTextChange: old = '      ', new = '		'
VsVim OnTextChange: old = '[367..373)', new = '[328..330)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 18
VsVim OnTextChange: old = '      ', new = '		'
VsVim OnTextChange: old = '[456..462)', new = '[413..415)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 19
VsVim OnTextChange: old = ' ', new = ''
VsVim OnTextChange: old = '[479..480)', new = '[432..432)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 20
VsVim OnTextChange: old = ' ', new = ''
VsVim OnTextChange: old = '[481..482)', new = '[433..433)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 21
VsVim OnTextChange: old = ' ', new = ''
VsVim OnTextChange: old = '[516..517)', new = '[467..467)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 22
VsVim OnTextChange: old = '      ', new = '		'
VsVim OnTextChange: old = '[521..527)', new = '[471..473)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 23
VsVim OnTextChange: old = '      ', new = '		'
VsVim OnTextChange: old = '[604..610)', new = '[550..552)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 24
VsVim OnTextChange: old = ' ', new = ''
VsVim OnTextChange: old = '[625..626)', new = '[567..567)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 25
VsVim OnTextChange: old = ' ', new = ''
VsVim OnTextChange: old = '[627..628)', new = '[568..568)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 26
VsVim OnTextChange: old = ' ', new = ''
VsVim OnTextChange: old = '[667..668)', new = '[607..607)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim OnTextChange: change 27
VsVim OnTextChange: old = '   ', new = '	'
VsVim OnTextChange: old = '[672..675)', new = '[611..612)'
VsVim OnTextChange: caret position = ?_v20_681_'<end>'
VsVim MarkGlyphTagger: Glyph Pairs
VsVim MarkGlyphTagger: 0 -> '^
VsVim MarkGlyphTagger::TextBufferChanged V21 (r21)
VsVim MarkGlyphTagger: Glyph Pairs
VsVim MarkGlyphTagger: 0 -> '^
VsVim AsyncTagger::GetTags 6 - 21
VsVim AsyncTagger Background New 6 - 21
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: tag 0 '^
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim MarkGlyphTagger::GetTags: starting...
VsVim VimBufferCoordinator::PreviewKeyUp V
VsVim VimKeyProcessor::KeyUp V Control
VsVim VimKeyProcessor::KeyUp Handled = False
VsVim VimBufferCoordinator::KeyUp V
VsVim VimBufferCoordinator::PreviewKeyUp LeftCtrl
VsVim VimKeyProcessor::KeyUp LeftCtrl None
VsVim VimKeyProcessor::KeyUp Handled = False
VsVim VimBufferCoordinator::KeyUp LeftCtrl
VsVim AsyncTagger::RaiseTagsChanged 0 - 22
VsVim AsyncTagger::GetTags 0 - 22
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus UserInput Key.Escape: 1496a755-94de-11d0-8c3f-00c04fc2aae2 103
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::Exec UserInput Key.Escape: 1496a755-94de-11d0-8c3f-00c04fc2aae2 103
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VimBufferCoordinator::PreviewKeyUp Return
VsVim VimKeyProcessor::KeyUp Return None
VsVim VimKeyProcessor::KeyUp Handled = False
VsVim VimBufferCoordinator::KeyUp Return
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
`

**Comment by amoldeshpande** [comment]
sorry the comment is long. i couldn't figure out if there is away to attach a file.

**Comment by ricksladkey** [comment]
@amoldeshpande Thanks, this is a big help!

Do you have some extension running that is automatically converting whitespace from spaces to tabs after you paste? I don't think anything in Visual Studio or VsVim supports that kind of feature and I can see that it is happening in the trace that you pasted.

If so, it's not a problem that it is doing that; it is just a part the conditions necessary to reproduce the bug.

**Comment by amoldeshpande** [comment]
I don't have any extensions that could obviously do this. The only thing I've installed myself is VsVim but VS installs a bunch of garbage extensions as well. None of them should be hooking into the editor though.

**Comment by amoldeshpande** [comment]
ooh, interesting. Apparrently visual studio decided to update my VsVim to 2.7.0.0 and I no longer get this behavior.  I can probably roll back if you need more information to track down the cause.

**Comment by ricksladkey** [comment]
@amoldeshpande, no, that's fine; don't downgrade. I have enough information now to repro with the old version and to check whether it indicates any problem with the current version. I will report back.

**Comment by ricksladkey** [comment]
OK, this particular bug was definitely already fixed in PR #2426 and so you shouldn't see the message box or any exception reported in the activity log with VsVim 2.7.0. I'm making some further changes to better report failures in this area and fix the other exception that I found.

**Comment by ricksladkey** [comment]
The other exception I found was pressing "enter" in normal mode while on the phantom line of the buffer when the stored caret column was non-zero.

**Comment by ricksladkey** [comment]
@amoldeshpande I suggest you turn off the VsVim output window now that we no longer need it. Quite a lot of output is generated and it could conceivable slow down VsVim. Thanks for your assistance reproducing the problem you were experiencing.

For the record, Visual Studio **does** have a feature whereby pasted code can be reformatted, spaces vs. tabs fixed, etc.  At least the C# language service does: "Automatically format on paste".

![image](https://user-images.githubusercontent.com/2680524/58771443-31d81c00-8582-11e9-87f2-08d09e93db59.png)

Oddly, I could not get this to work when I pasted using Edit -> Paste, but, for the purposes of reproducing the multi-change edit that it produces, I found that I could use Edit -> Advanced -> Format Document, which has the same net effect.
